### PR TITLE
api: Update API version to 2.5.0

### PR DESCRIPTION
**Pull request by marclop**
## Description

Updates the API version to match the spec version

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       id: go
 
     - name: Check out code into the Go module directory
-      uses: actions/checkout@v1
+      uses: actions/checkout@v2
```
Comment (ppapapetrou76) [inline .github/workflows/go.yml]: ??? is this related to api version? 😄

Comment (marclop) [reply .github/workflows/go.yml]: It is not, but it's necessary to make the build pass, unfortunately it seems that there's some bug which was fixed in v2+ and now the checkout v1 versions fail. (see the other open PR too)
